src/nunit.analyzers.tests/ConstActualValueUsage/ConstActualValueUsageAnalyzerTests.cs

```diff
@@ -128,11 +129,37 @@ public void Test()
             AnalyzerAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
         }
 
+        [Test]
+        public void AnalyzeWhenStringEmptyArgumentIsProvidedForAreEqual()
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+                public void Test()
+                {
+                    string actual = ""act"";
+                    Assert.AreEqual(actual, string.Empty);
+                }");
+
+            AnalyzerAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
+        }
+
+        [Test]
+        public void AnalyzeWhenStringEmptyArgumentIsProvidedForAssertThat()
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+                public void Test()
+                {
+                    string actual = ""act"";
+                    Assert.That(string.Empty, Is.EqualTo(actual));
```

Comment:
```suggestion
                    string expected = ""exp"";
                    Assert.That(↓string.Empty, Is.EqualTo(expected));
```

Reply:
No, string.Empty is the expected value. The code was supposed to test if an actual value is string.Empty. As the arguments are reversed, this should trigger an analyzer violation. String.Empty is treated as if it is a constant.
One wouldn't write this code, but it could be the result of the CodeFix to convert Classic AreEqual into Constraint model.

Reply:
Similar comment as above. The signature of `Assert.That` is `Assert.That<TActual>(TActual actual, IResolveConstraint expression)`, so the first argument is the actual value and the last is the expected value expressed as a constraint.